### DISCUSSION TITLE
Can we do "Linked Duplicate Frames"?

### DISCUSSION THREAD

**Discussion by konero** [keiwando/toonsquid#626] [General]
Hi, I've just been learning ToonSquid (I work professionally as an animator, not had too much time with it yet) but my question is can we do "linked duplicates" for frames? In traditional animation this is called "instancing", if frames were numbered we can do [1, 2, 3, 2, 3, 2] for example on an x-sheet, and if I modify [2] then anywhere [2] is instanced is modified and updated.

In Clip Studio Paint, we right-click on the timeline at the positions we want to instance the drawing layer, then choose the layer we want instanced and in OpenToonz we can just type the number we want on the x-sheet (drawings are stored in "levels" with each frame being given a number), very straightforward. Other apps like DigiCel FlipBook and Animation Paper you have to copy-paste. And there's others like Callipeg where we create "linked duplicates" from a popup menu, etc.

Symbols would be a tedious workaround for me, so I'm hoping there's another way to do it.

Thanks in advance!

**Comment by Cat-Butler** [comment]
I can't think of a way to do it directly in ToonSquid without symbols.
Unless somebody smarter than me (not difficult) comes up with a better solution, I think you're going to have to copy-paste in ToonSquid too (if they're close enough together and you have Magnetic Mode, then you also have the option to Duplicate-and-drag). 

However, you can get most of the way there by combining this with symbols (hear me out, we can avoid at least part of the tedious aspects): If you tap on the drawing frame in the timeline and choose to convert it to a symbol, it remains in place and you can still copy-paste as if were a normal drawing frame, but you can then edit the symbol to change it in all the frames it was copied to (plus the original, of course). I admit this might not work for all contexts (for example, keyframed multi-layered drawings that show as a single layer in the timeline).

**Reply by otaworks** [reply]
I wish you could try out Clip Studio Paint for yourself to see how the feature described works. You can try the app out for 1-3 months for free on your iPad, and if you do, get the EX version. I'll link to a video that hopefully touches on that feature:
https://www.youtube.com/watch?v=ZG1ITZxD_9I&pp=ygUGZGVycmVr

**Reply by Cat-Butler** [reply]
Thanks for the good wishes, but I much prefer the rigging approach (came to ToonSquid from Moho, originally from 3D rigging).

I can understand the feature readily enough (although I clearly hadn't given enough thought to the clean-up/inbetween stages, I was mostly thinking about final frames and timing/editing), and can see the appeal for frame-by-frame animation, so I hope the feature happens (although I'm guessing there probably won't be much chance of that until 2.0 beta finalises - and the 2.0 beta proposed feature list looks like it will be very much appeal to my rigging preferences if the features make it to release).

I used to use CSP on the desktop for a while as an illustration app (Affinity Designer is my go-to app these days, I've always been a big fan of vector-based workflows) - it's an excellent illustration/cartooning app, but I wasn't impressed with its early animation implementation (I believe that has improved somewhat).
Much of that was probably my very rigging-centric animation preference, I've heard good things about it from people using more of a frame-by-frame approach.
(Was also not impressed by the early CSP iPad implementation interface, which I did trial, although I believe that has improved significantly.)

**Comment by budaloops** [comment]
I don’t have an answer to offer here now, but it looks like you’ve been exploring different iPad animation apps. Are you considering the iPad as your main animation tool? And you can always combine apps and be one of the first in your lane!

**Comment by konero** [comment]
Sorry, late response from me! Thank you for the suggestions however it seems it would be tedious to use and manage symbols for this task, especially when doing cleanup or inbetween. I really hope to see linked-duplicates with these objects shown with some visual cue so we can tell them apart.

**Reply by Cat-Butler** [reply]
Fair enough, I was thinking more about final frames rather than clean-up/inbetweens.
I prefer a rigging approach, so I had not considered the frame-by-frame approach in enough depth.

I can see the appeal for frame-by-frame style animation, so I hope it happens.

It looks like maybe the framework is there, so implementation of this type of in-timeline-symbol-editing (integrating with onion skins, etc) might be possible, although I wouldn't like to guess how difficult the implementation would be (and I speculate it won't happen until after the 2.0 beta features are finalised to a release).

**Comment by otaworks** [comment]
Yeah, this would be amazing to have! I also use CSP on my iPad to animate, so I understand completely. It helps save a lot of time when reusing frames, so it would be a great feature. I can't believe I never thought of TS not having that feature. Then again, I mainly use Clip Studio Paint in professional productions, so this never occurred to me.

Hopefully this feature is considered
